### PR TITLE
Add ability to set a MAP as optional via tags

### LINKED ISSUE #280: How to make an optional map?
Take the sample below:
```go
type Spans struct {
  Attributes map[string]string
}

s := parquet.SchemaOf(new(RowType))
fmt.Println(s)
```

It prints:
```
message Spans {
  required group Attributes (MAP) {
    repeated group key_value {
      required binary key (STRING);
      required binary value (STRING);
    }
  }
}
```

However, I want the Attributes map to be optional:
```
message Spans {
  optional group Attributes (MAP) {
    repeated group key_value {
      required binary key (STRING);
      required binary value (STRING);
    }
  }
}
```

I try to add a tag, but it makes the value optional, not the map.
```
type Spans struct {
	Attributes map[string]string  `parquet:"Attributes,optional"`
}
```

Could you please teach me how to do it?

### PR DESCRIPTION
**Pull request by Pryz**
Fixes #280 

This change update the schema API with the following:
- The tag `parquet-value` is introduce and allows the configuration of the MAP values.
- The tag `parquet` will now configure the map (top level) itself.

### CODE REVIEW

**Review by kevinburkesegment — APPROVED** [review]
Do we need to do anything to indicate to callers now that an empty Group struct is not usable?

File: node.go [hunk 1]
```diff
@@ -235,13 +235,56 @@ func fieldRepetitionTypeOf(node Node) format.FieldRepetitionType {
 	}
 }
 
-type Group map[string]Node
+type Group struct {
+	nodes    map[string]Node
+	optional bool
+}
```
Comment (achille-roussel) [inline node.go:241]: Instead of changing `Group`, would it work if we used the `Optional` wrapper?
```go
Optional(Group{ ... })
```

Comment (Pryz) [reply node.go:241]: mmm will take a look but we are gonna have a type issues with `mapNode` which is expecting a Group and not a Node right ?

Comment (Pryz) [reply node.go:241]: Actually we can vary the tag there. Thanks ! Will refactor :)

**Pull request closed without merging**